runtime/src/snapshot_utils.rs

```diff
@@ -145,11 +159,11 @@ impl IncrementalSnapshotArchiveInfo {
         &self.inner.slot
     }
 
-    fn _hash(&self) -> &Hash {
+    fn hash(&self) -> &Hash {
         &self.inner.hash
     }
 
-    fn _archive_format(&self) -> &ArchiveFormat {
+    fn archive_format(&self) -> &ArchiveFormat {
```

Comment:
Side note, I think a good way to reduce this code duplication between `FullSnapshotArchiveInfo` and `IncrementalSnapshotArchiveInfo` is to create a trait `SnapshotArchiveInfo` like so:

```
trait SnapshotArchiveInfo {
   // constructor 
    fn from_snapshot_archive_info(snapshot_archive_info: SnapshotArchiveInfo) -> Self;
    // getter
    fn snapshot_archive_info(&self) -> &SnapshotArchiveInfo;

  // default methods to avoid duplication
    fn hash(&self) -> &Hash {
        &self.snapshot_archive_info().hash
    }

    fn archive_format(&self) -> &ArchiveFormat {
      &self.snapshot_archive_info().archive_format
      }
}

```

Reply:
I like this idea! One thing is that the `IncrementalSnapshotArchiveInfo` needs a `base_slot` parameter too. Do you know if there's a way to handle this within the Trait?

Reply:
I'm implemented the trait but _without_ a constructor. I think for now that should be good.

The next thing I'm working on is adding in incremental snapshot support to the background services (AccountsBackgroundService, AcccountsHashVerifier, SnapshotPackagerService). I'll by building out the SnapshotArchiveInfo more. In particular, moving Full and Incremental under a single enum, so that way when a background service makes a package, it'll contain the info about full vs incremental, and it'll carry through to the next service.

So yeah, is this OK for now? I did change `slot()` and `archive_format()` to not return references anymore, since they are `Copy` and small, and always used as values anyway.